tests/functional/ext/docparams/raise/missing_raises_doc.py

```diff
@@ -11,7 +11,7 @@ def test_ignores_no_docstring(self):
     raise RuntimeError("hi")
 
 
-def test_ignores_unknown_style(self):
+def test_ignores_unknown_style(self):  # [missing-raises-doc]
```

Comment:
This is the "breaking change". When we don't recognize the docstring style and accept-no-raises-doc is False, we now raise a missing-raises-doc.

Reply:
I'm not sure we should introduce a breaking change. Can we fix this without doing so?

Reply:
I believe the issue is that my suggestion was not correct, and always suppressed missing-raises-doc when accept-no-raises-doc was set.

A better way to handle this option would be to suppress missing-raises-doc when there are no matching sections (as was the case previously), or (and this would be new) when no exceptions are referenced in the docstring.

Below is a possible way to do both. It allows reverting all tests to their previous behavior with the default accept-no-raises-doc value.

~~~diff
diff --git a/pylint/extensions/docparams.py b/pylint/extensions/docparams.py
index 936cda683..c789a8472 100644
--- a/pylint/extensions/docparams.py
+++ b/pylint/extensions/docparams.py
@@ -274,8 +274,6 @@ class DocstringParameterChecker(BaseChecker):
             self.add_message("redundant-yields-doc", node=node)
 
     def visit_raise(self, node: nodes.Raise) -> None:
-        if self.linter.config.accept_no_raise_doc:
-            return
         func_node = node.frame(future=True)
         if not isinstance(func_node, astroid.FunctionDef):
             return
@@ -295,6 +293,10 @@ class DocstringParameterChecker(BaseChecker):
         doc = utils.docstringify(
             func_node.doc_node, self.linter.config.default_docstring_type
         )
+
+        if self.linter.config.accept_no_raise_doc and not doc.exceptions():
+            return
+
         if not doc.matching_sections():
             if doc.doc:
                 missing = {exc.name for exc in expected_excs}
~~~

Reply:
@bchardin I tried your change but there's not difference in the output afaik ? I(m going to investigate the issue.